packages/assert/src/assert.js

```diff
@@ -57,6 +57,24 @@ export { quote as q };
 
 export { makeAssert };
 
+/**
+ * @template T
+ * @param {T | null | undefined} val
+ * @param {string} [message]
+ * @returns {T}
+ */
+export function assertDefined(val, message) {
```

Comment:
@erights WDYT of this? similar to `Nat` but for a nullish check. See below for an example in action.